libsql-server/src/lib.rs

```diff
@@ -726,3 +735,68 @@ where
         }
     }
 }
+
+/// Setup sqlite to use the same allocator as sqld.
+/// the size of the allocation is stored as a usize before the returned pointer. A i32 would be
+/// sufficient, but we need the returned pointer to be aligned to 8
+fn setup_sqlite_alloc() {
+    use std::alloc::GlobalAlloc;
+
+    unsafe extern "C" fn malloc(size: i32) -> *mut c_void {
+        let size_total = size as usize + size_of::<usize>();
+        let layout = Layout::from_size_align(size_total, align_of::<usize>()).unwrap();
+        let ptr = GLOBAL.alloc(layout);
+        *(ptr as *mut usize) = size as usize;
```

Comment:
Allocator can fail and return null - in this case `SEGFAULT` will happen here (although I hope that SQLite can handle out of memory conditions in some cases). Should `malloc` be implemented more carefully then?